### PR TITLE
Fixes from bugbash

### PR DESCRIPTION
**Pull request by arsing**
- Change config.toml.template from `r--` to `rw-`

  This allows the user's editor to write it in-place instead of overwriting it.

- Enable logging for `hyper` via `env_logger`.

- Disable HTTP/2 via ALPN.

  This is a temporary workaround until we figure out why IS can't talk HTTP/2
  to Edge Hub even after enabling the `http2` feature of `hyper`.

- Parse DPS global endpoint as a `url::Url` so that invalid values
  are rejected at config parse time.

- Allow duplicate principals and merge them at runtime.

### CODE REVIEW

**Review by daprilik — APPROVED** [review]

File: contrib/centos/aziot-identity-service.spec [hunk 1]
```diff
@@ -174,7 +174,7 @@ fi
 %attr(400, aziottpm, aziottpm) %{_sysconfdir}/aziot/tpmd/config.toml.default
 %attr(700, aziottpm, aziottpm) %dir %{_sysconfdir}/aziot/tpmd/config.d
 
-%attr(400, root, root) %{_sysconfdir}/aziot/config.toml.template
+%attr(700, root, root) %{_sysconfdir}/aziot/config.toml.template
```
Comment (daprilik) [inline contrib/centos/aziot-identity-service.spec:177]: Wait, shouldn't this be 600?

**Comment by gordonwang0** [comment]
Regarding the duplicate principals:

I think it's better to allow `iotedge config apply` to generate a principals list with duplicate uids and modify the `principal_to_map` function (in keyd/certd lib.rs) to dedup.

**Review by gordonwang0 — CHANGES_REQUESTED** [review]
See comment above

**Comment by arsing** [comment]
True, that would make patching easier too.

File: contrib/centos/aziot-identity-service.spec (hunk 1 above)
Comment (arsing) [reply contrib/centos/aziot-identity-service.spec:177]: Ah yes, I fixed it in iotedge but forgot to fix it here. Fixed now.

**Review by gordonwang0 — APPROVED** [review]

**Pull request merged**